### DISCUSSION TITLE
How to prevent setValue from rerendering the whole form for field registered dynamically?

### DISCUSSION THREAD

**Discussion by Drakota** [react-hook-form/react-hook-form#1980] [Q&A] [answered]
Hi everyone, I'm trying to use setValue and register inside dynamically rendered elements, [I've made a quick codesandbox to illustrate my issue](https://codesandbox.io/s/amazing-rgb-6rpot?file=/src/Setting.js:519-524). I have an input inside a component that registers a custom made input that uses setValue and uses that value elsewhere in the same component. My codesandbox works great and when I submit my form I do have access to all my values, but the problem is when enabling `Highlight updates when components render.` like this:
![image](https://user-images.githubusercontent.com/22665836/85934773-e5126200-b8b5-11ea-97ce-b99d5af06bb9.png)
I've noticed that the whole form is rerendering when setValue is called for any input, I was wondering if there was a way to achieve what I'm trying to do without rerendering the whole form, only the component that is watching what's modified.

**Comment by bluebill1049** [comment, marked as answer]
The re-render is caused by your watch `const item = watch(`${id}`, {});`

**Reply by Drakota** [reply]
Sure, but I need the watch to display something when it's modified. In my example, I'm modifying my input and I want it to display the text.

**Reply by bluebill1049** [reply]
then you will have to use `useWatch` which will be released as part of V6, `watch` will trigger re-render at form or hook level.

**Reply by Drakota** [reply]
Looks promising will check it out in rc, thank you!
